src/main/kotlin/org/rust/openapiext/psi.kt

```diff
@@ -96,14 +98,23 @@ private class RsWithMacrosRecursiveElementWalkingVisitor(
         }
     }
 
-    private fun processMacro(element: RsMacroCall) {
+    private fun processMacro(element: RsMacroCall): Boolean {
         if (nestedMacroDepth > DEFAULT_RECURSION_LIMIT) {
-            return
+            return true
         }
-        val expansion = element.expansion ?: return
+        if (element.macroArgumentAny == null) return false
+        val expansion = element.expansion
+        // builtin macro without expansion => we should use `super.visitElement`
+        // usual macro without expansion => skip it
+            ?: return element.resolveToMacro()?.hasRustcBuiltinMacro != true
         for (expandedElement in expansion.elements) {
             val visitor = RsWithMacrosRecursiveElementWalkingVisitor(processor, nestedMacroDepth + 1)
             expandedElement.accept(visitor)
         }
+        return true
     }
 }
+
+private val RsMacroCall.macroArgumentAny: RsElement?
```

Comment:
This can be confused with `RsMacroCall.macroArgumentElement`. Let's choose another name or use `macroArgumentElement` (if reasonable here)